src/test/e2e/api/admin/project/features.e2e.test.ts

```diff
@@ -3153,3 +3157,52 @@ test('Enabling a feature environment should add the default strategy when only d
             expect(res.body.strategies[1].disabled).toBeFalsy();
         });
 });
+
+test('Updating feature strategy sort-order should trigger a FeatureStrategyUpdatedEvent', async () => {
+    const envName = 'sort-order-within-environment-one';
+    const featureName = 'feature.sort.order.event.list';
+    // Create environment
+    await db.stores.environmentStore.create({
+        name: envName,
+        type: 'test',
+    });
+    // Connect environment to project
+    await app.request
+        .post('/api/admin/projects/default/environments')
+        .send({
+            environment: envName,
+        })
+        .expect(200);
+
+    await app.request
+        .post('/api/admin/projects/default/features')
+        .send({ name: featureName })
+        .expect(201);
+
+    await addStrategies(featureName, envName);
+    const { body } = await app.request.get(
+        `/api/admin/projects/default/features/${featureName}/environments/${envName}/strategies`,
+    );
+
+    const strategies: FeatureStrategySchema[] = body;
+    let order = 1;
+    const sortOrders: SetStrategySortOrderSchema = [];
+
+    strategies.forEach((strategy) => {
+        sortOrders.push({ id: strategy.id!, sortOrder: order++ });
+    });
+
+    await app.request
+        .post(
+            `/api/admin/projects/default/features/${featureName}/environments/${envName}/strategies/set-sort-order`,
+        )
+        .send(sortOrders);
+    // .expect()
+    const response = await app.request.get(`/api/admin/events`);
+    const { body: eventsBody } = response;
+    let { events } = eventsBody;
+
+    for (let i = 0; i < strategies.length; i++) {
```

Comment:
I'd use manual indexes and assertions since for loops break the declarative nature of the tests (e.g. people have to mentally process i++)

Reply:
👍